spec/system/user_page/user_preferences_interface_spec.rb

```diff
@@ -31,4 +31,34 @@
       end
     end
   end
+
+  describe "Default Home Page" do
+    context "when a user has picked a home page that is no longer available in top_menu" do
+      it "shows the selected homepage" do
+        SiteSetting.top_menu = "latest|hot"
+
+        user.user_option.update!(homepage_id: UserOption::HOMEPAGES.key("unread"))
+        user_preferences_page.visit(user)
+        click_link(I18n.t("js.user.preferences_nav.interface"))
+
+        dropdown = PageObjects::Components::SelectKit.new("#home-selector")
+
+        expect(dropdown).to have_selected_name("Unread")
+      end
+    end
+
+    it "shows only the available home pages from top_menu" do
+      SiteSetting.top_menu = "latest|hot"
+
+      user_preferences_page.visit(user)
+      click_link(I18n.t("js.user.preferences_nav.interface"))
```

Comment:
same

Reply:
Good idea, but it seems to be flakey locally with `user_preferences_interface_page.visit(user)`. Probably because the test manipulates the user options just before visiting the page. 

I'll leave it as is in order to avoid introducing flakiness.